src/main/kotlin/step2/Caculator.kt

```diff
@@ -0,0 +1,75 @@
+package step2
+
+import java.util.LinkedList
+import java.util.Queue
+
+object Caculator {
+
+    /**
+     * Calculates if the [input] is a correct mathematical expression.
+     *
+     * @param input String of mathematical expression.
+     * @throws IllegalArgumentException If the [input] is not a correct mathematical expression.
+     */
+    fun calculate(input: String?): Double {
+        val expression: Queue<Token> = input.toExpression()
+
+        var result: Double = 0.0
+        var curSymbol: Token.Symbol? = null
```

Comment:
```suggestion
        val expression = input.toExpression()

        var result = 0.0
        var curSymbol = null
```

타입 추론을 사용해볼까요? 타입 추론은 코틀린의 장점 중에 하나라고 생각해요. 저에게 타입 추론은 메서드와 변수명을 더 명확하게 만들어준다고 생각이 들어요. 또한 코드를 더 간결하게 만들어준다고 생각해요.

Reply:
저는 타입을 명시해주지 않으면 깃헙으로 내용을 볼 때 타입을 유추하기 힘들어서 써주는 쪽으로 해왔는데요~
플람 의견에도 공감합니다👍